### PR TITLE
Refactor GraphQL error messages

### PR DESCRIPTION
**Pull request by DimuthuMadushan**
## Purpose

Fixes: [#2076](https://github.com/ballerina-platform/ballerina-standard-library/issues/2076)

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: ballerina/common_utils.bal [hunk 1]
```diff
@@ -228,3 +238,102 @@ isolated function getListMemberTypeFromType(__Type argType) returns __Type {
...
+        if argValue.length() > i+1 {
+            listItems += ", ";
+        }
+    }
+    listItems += "]";
+    return listItems;
+}
+
+isolated function appendScalarValues(parser:ArgumentValue argValue, string errorValue = "") returns string {
+    string scalarValue = errorValue;
```
Comment (ThisaruGuruge) [inline ballerina/common_utils.bal:315]: This variable too is redundant. We can just return the value.

File: ballerina/common_utils.bal [hunk 2]
```diff
@@ -228,3 +238,102 @@ isolated function getListMemberTypeFromType(__Type argType) returns __Type {
...
+        scalarValue += "null";
+    }
+    return scalarValue;
+}
+
+isolated function getInputObjectFieldFormPath((string|int)[] path, string name) returns string {
+    int? index = path.indexOf(name);
+    string inputObjectField = name;
+    if index is int {
+        if index > 0 {
```
Comment (ThisaruGuruge) [inline ballerina/common_utils.bal:330]: What if the index is `0`?

File: ballerina/tests/resources/expected_results/enum_type_default_value_with_string_literal.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
     "errors": [
         {
-            "message": "Variable \"day\" of type \"Weekday\" has invalid default value: MONDAY. Expected type \"Weekday\", found MONDAY",
+            "message": "Variable day expected value of type \"Weekday\", found MONDAY",
```
Comment (ThisaruGuruge) [inline ballerina/tests/resources/expected_results/enum_type_default_value_with_string_literal.json:4]: Shouldn't we have double quotes for the variable name? Is the double quotes around the type name necessary?

File: ballerina/common_utils.bal (hunk 2 above)
Comment (DimuthuMadushan) [reply ballerina/common_utils.bal:315]: return the `name`.

File: ballerina/tests/resources/expected_results/enum_type_default_value_with_string_literal.json (hunk 3 above)
Comment (DimuthuMadushan) [reply ballerina/tests/resources/expected_results/enum_type_default_value_with_string_literal.json]: both are needed.

File: ballerina/common_utils.bal [hunk 4]
```diff
@@ -228,3 +228,96 @@ isolated function getListMemberTypeFromType(__Type argType) returns __Type {
...
+                inputFields += string`${value.getName()}: "${fieldValue}"`;
+            } else if fieldValue is () {
+                inputFields += string`${value.getName()}: null`;
+            } else if fieldValue is Scalar {
+                inputFields += string`${value.getName()}: ${fieldValue}`;
+            }
+        } else {
+            inputFields = appendScalarValues(value, inputFields);
+        }
+        if argValue.length() > i+1 {
```
Comment (ThisaruGuruge) [inline ballerina/common_utils.bal:268]:
```suggestion
        if argValue.length() > i + 1 {
```

File: ballerina/common_utils.bal [hunk 5]
```diff
@@ -228,3 +228,96 @@ isolated function getListMemberTypeFromType(__Type argType) returns __Type {
...
+                listItems += string`"${listItemValue}"`;
+            } else if listItemValue is () {
+                listItems += "null";
+            } else if listItemValue is Scalar {
+                listItems += string`${listItemValue}`;
+            }
+        } else {
+            listItems = appendScalarValues(value, listItems);
+        }
+        if argValue.length() > i+1 {
```
Comment (ThisaruGuruge) [inline ballerina/common_utils.bal:295]:
```suggestion
        if argValue.length() > i + 1 {
```

File: ballerina/tests/resources/expected_results/list_type_with_invalid_variable_default_values_1.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
     "errors": [
         {
-            "message": "Variable \"words: In element #3:\" of type \"String\" has invalid default value: Ballerina. Expected type \"String\", found Ballerina",
+            "message":"Variable \"words: In element #3:\" expected value of type \"String\", found Ballerina",
```
Comment (ThisaruGuruge) [inline ballerina/tests/resources/expected_results/list_type_with_invalid_variable_default_values_1.json:4]:
```suggestion
            "message": "Variable \"words: In element #3:\" expected value of type \"String\", found Ballerina",
```

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**